packages/openapi-typescript/src/load.ts

```diff
@@ -296,7 +296,19 @@ export default async function load(schema: URL | Subschema | Readable, options:
       walk(options.schemas[k].schema, (rawNode, nodePath) => {
         const node = rawNode as unknown as SchemaObject;
         if (!node.discriminator) return;
-        options.discriminators[schemaID === "." ? makeTSIndex(nodePath) : makeTSIndex(["external", k, ...nodePath])] = node.discriminator;
+        const discriminator: DiscriminatorObject = { ...node.discriminator };
+
+        // handle child oneOf types (mapping isn’t explicit from children)
+        const oneOf: string[] = [];
+        if (Array.isArray(node.oneOf)) {
+          for (const child of node.oneOf) {
+            if (!child || typeof child !== "object" || !("$ref" in child)) continue;
+            oneOf.push(child.$ref);
```

Comment:
This adds automatic inference from parent `oneOf` types, but unfortunately the mapping is brittle (probably breaks for remote schemas).

I think the only way to fix this is if this library pre-bundled schemas via something from [Redocly](https://redocly.com/)